src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Integer.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    internal ref partial struct CborReader
+    {
+        // Implements major type 0 decoding per https://tools.ietf.org/html/rfc7049#section-2.1
+        public ulong ReadUInt64()
+        {
+            ReadInitialByte(out CborDataItem header);
+
+            switch (header.MajorType)
+            {
+                case CborMajorType.UnsignedInteger:
+                    ulong value = ReadUnsignedInteger(in header, out int additionalBytes);
+                    AdvanceBuffer(1 + additionalBytes);
+                    return value;
+
+                default:
+                    throw new ArgumentException("Data item type mismatch");
```

Comment:
InvalidOperationException is more appropriate (there's no argument here that's wrong, the call is inappropriate to the state).